clients/webclient/src/components/Users/CustomerEdit.vue

```diff
@@ -0,0 +1,139 @@
+                    <v-text-field
+                        :rules="[numbers, phoneNoLength]"
+                        label="Telefonnummer"
+                        v-model="phone"
+                        required
+                    ></v-text-field>
+                </v-col>
+                <v-col>
+                    <v-text-field
+                        label="E-post"
+                        v-model="email"
+                        required
+                    ></v-text-field>
+                </v-col>
+            </v-row>
+            <v-row>
+                <v-col :cols="2">
+                    <p class="font-weight-black">
+                        <label for="allergylist">Mine allergier</label>
+                    </p>
+                </v-col>
+                <v-col>
+                    <p
+                        class="font-weight-light"
+                        v-for="allergy in allergies"
+                        :key="allergy"
+                        id="allergylist"
+                    >
+                        {{ allergy }}
+                    </p>
+                </v-col>
+            </v-row>
+            <v-row>
+                <v-col>
+                    <v-btn @click="sendToDb" color="primary" class="ma-1">
+                        Send inn
+                    </v-btn>
+                    <v-btn color="secondary" @click="cancel" class="ma-1">
+                        Avbryt
+                    </v-btn>
+                </v-col>
+            </v-row>
+        </v-container>
+    </v-form>
+</template>
+
+<script lang="ts">
+import Vue from "vue";
+import Component from "vue-class-component";
+import { Prop } from "vue-property-decorator";
+import api from "../../api/api";
+import { Userprofile } from "../../../../../server/src/interfaces";
+import CustomerProfile from "../Users/CustomerProfile.vue";
+
+@Component({
+    components: {
+        CustomerProfile,
+    },
+})
+export default class CustomerEdit extends Vue {
+    @Prop() loggedInUser!: string;
+    @Prop() userprofile!: Userprofile;
+    private fullname = "";
+    private address = "";
+    private phone = "";
+    private email = "";
+    private allergies: string[] = [];
+
+    async fillForm() {
+        let user = await api.getUserprofile();
+        if (user) {
+            this.fullname = user.fullname;
+            this.address = user.address;
+            this.phone = user.phone;
+            this.email = user.email;
+            this.allergies = user.allergies;
+            console.log(this.userprofile);
+        }
+    }
+
+    async beforeMount() {
+        this.fillForm();
+    }
+
+    numbers(value: string) {
+        return (
+            !isNaN(parseInt(value)) || "Vennligst oppgi et gyldig postnummer"
+        );
+    }
+
+    phoneNoLength(value: string) {
+        return value.length >= 8 || "Vennligst oppgi et gyldig telefonnummer";
+    }
+
+    async sendToDb() {
+        let updateUserprofile = {
```

Comment:
Dette objektet trengs ikke, dere kan bruke proppen som vil være oppdatert: await api.putUserprofile(this.userprofile);